derp/derphttp/derphttp_client.go

```diff
@@ -513,6 +517,11 @@ func (c *Client) SetURLDialer(dialer func(ctx context.Context, network, addr str
 	c.dialer = dialer
 }
 
+// SetNodeDialer sets the dialer to use for dialing DERP nodes.
+func (c *Client) SetNodeDialer(dialer func(ctx context.Context, node *tailcfg.DERPNode) (net.Conn, error)) {
+	c.nodeDialer = dialer
```

Comment:
Do we need to protect via mutex or is this invoked synchronously at a safe time?

Reply:
It's invoked synchronously, but I'll add a mutex to be safe.